src/pydata_sphinx_theme/theme/pydata_sphinx_theme/components/navbar-logo.html

```diff
@@ -14,10 +14,17 @@
   {% set image_dark = image_dark if image_dark.startswith("http") else pathto('_static/' + image_dark, 1) %}
   {% set alt = theme_logo.get("alt_text", "Logo image") %}
   {% if is_logo %}
-    <img src="{{ image_light }}"
-         class="logo__image only-light"
-         alt="{{ alt }}"/>
-    <img src="{{ image_dark }}" class="logo__image only-dark" alt="{{ alt }}"/>
+    {# Theme switching is only available when JavaScript is enabled.
+     # Thus we should add the extra image using JavaScript, defaulting
+     # depending on the value of default_mode; and light if unset.
+     #}
+    {% if default_mode == "dark" %}
```

Comment:
I think you can avoid the tags duplication something like: 

```html
{% set js_mode = "light" if default_mode == "dark" else "dark" %}
<img src="{{ image[default_mode] }}" class="logo__image only-{{ default_mode}}" alt="{{ alt }}"/>
<script>document.write(`<img src="{{ image[js_mode] }}" class="logo__image only-{{ js_mode }}" alt="{{ alt }}"/>`);</script>
```

it would also require to change image into a dictionnary (maybe wait for https://github.com/pydata/pydata-sphinx-theme/pull/1132 to be merged ?)

Reply:
> from the general point of view, what would you think if default_theme is set to "light" for all documentation by default

That was my assumption while modifying the CSS or the logo was to fallback on “light”. I guess you know the theme better to evaluate how safe it is.

Reply:
But wait, wouldn’t it shift the default behavior from `auto` to `light`?

Reply:
you are very right, let's leave it as it is

Reply:
@lunar-debian, #1132 have been merged in main, could you adapt your code as suggested in my first comment ?